src/engraving/libmscore/barline.cpp

```diff
@@ -1503,18 +1505,8 @@ bool BarLine::setProperty(Pid id, const PropertyValue& v)
 void BarLine::undoChangeProperty(Pid id, const PropertyValue& v, PropertyFlags ps)
 {
     if (id == Pid::BARLINE_TYPE && segment()) {
-        const BarLine* bl = this;
         BarLineType blType = v.value<BarLineType>();
-        if (blType == BarLineType::START_REPEAT) {     // change next measures endBarLine
-            if (bl->measure()->nextMeasure()) {
-                bl = bl->measure()->nextMeasure()->endBarLine();
-            } else {
-                bl = 0;
-            }
```

Comment:
I literally have no idea why this code was here...

Reply:
Hard to see how that issue could have anything to do with barlines at all!

Reply:
Ooops, sorry, I had the wrong link - fixed now.

Reply:
Ah ok.  But it is rather surprising 3.6 is still so broken in this regard - there's literally no way I can see of making the first barline in a score a "start repeat" using just the inspector, unless you insert a dummy measure at the beginning first.

Reply:
The Inspector was never intended to be a supported way of creating repeat barlines in the first place.  And in fact, it's only possible for single-staff scores - scores with multiple staves gray this option out, because the Inspector changes only the current staff (in MU3 anyhow) and it would be inconsistent to have a repeat on one staff but not another.

So, between the fact that it's not actually the officially supported way of creating repeats (the palette is), and it isn't possible to do via the Inspector at all except for single-staff scores, and of course it has had this obvious bug for the past couple of years but only two or three reports, it seems pretty safe to say, most people don't create repeats this way to begin with.

Reply:
Actually I've performed off scores that had repeat marks only in two staves,  while the rest of the ensemble played through composed music. I wouldn't necessarily recommend it, but it can work. I assume there'd be some way to kludge it...

Reply:
Right, it definitely can happen that you have scores with mismatched repeats like this, but we didn't want it to be too easy to create that accidentally, which is why it's disabled in the Inspector.  The current workaround would be to add the repeat barline from the Symbols palette and position it manually.  Definitely room for improvement.

Offhand I can't think of a good reason to keep the repeat options in the Inspector given that they are only active for single-staff scores and in that case they don't do anything differently from the palette, aside from appearing in the wrong measure.

Also, in case it isn't obvious: normally, the Inspector in MU3 applies a barline change to just that staff, similar to using Ctrl+click from the palette.  Ctrl is our standard shortcut to mean "apply this to a single staff only even though it would normally affect all".  But Ctrl+click on the repeat barlines still adds to all staves.

Reply:
Well, they work in MS4 across staves. If it could just be made consistent and not cause crashes, then that's an improvement over 3. However, if it is rife with issues, then I'd potentially remove it from the Inspector.

All in all - if it works, keep it. Don't limit it to one instrument. That doesn't make sense to the user.

Agreed about manual placement of symbols to achieve single instrument repeats in a score with multiple instruments.

Reply:
I'm mostly in agreement.  But I would just observe, it *is* important to allow the user to set barline properties for just a single staff.  It's the key to how you can override barline spans, or hide barlines for just one staff, or create "mensurstrich" etc.  Quite a few real-world effects require modifying barlines properties independently per staff.

Luckily most of the settings in the MU4 properties panel do continue to work that way - it seems only the "Style" has been modified to affect all staves.  To me that's a mistake - only one staff's barline is selected, so only that should be affected.  But it would be fine if clicking a barline actually selected across all staves like we do for key and time signatures.  And then Ctrl+click could still be used to select just a single staff's barline and thus affect it's properties.  That to me would be more consistent and logical without giving up functionality.  Only issue would be the discoverability of Ctrl+click as a method of limiting selection to a single staff, which is definitely a consideration and a reason why things are as they are in MU3.

Reply:
So bottom line for me - right now Style works differently from the other barline properties.  It does nothing different from the palette except impose limitations and bugs.  I think the best course of action is to remove this.  Leaving the other properties as they are - affecting only the current staff.

Reply:
Removing the repeat options from the properties panel would still require code changes to support what happens when you do select a repeat barline - in that case you wouldn't want to show any option to charge the barline type (only height/ span etc.). But honestly I think what I've implemented here works fine,  unless QA find some specific issues,  there are definitely some cases that mightn't work as expected.

Reply:
I just tried with v4 *and* v3, using CTRL+click to select a barline for just a single staff doesn't work - in fact, it looks the same as regular click (only the barline on the staff selected is highlighted), but any property changes apply to all staves. So how am I supposed to change the barline style (e.g. just to 'dotted') for a single staff using properties panel/inspector?

Reply:
Not sure which properties you tried, but in both MU3 and MU4, all properties I tried affect just the selected staff *except* for Style, which affects them all.  Eg, try toggling visibility or any of the span options or setting an X offset.  So right now it's inconsistent.  Most properties affect only what's selected, except Style, which affects that barline across all staves.

And yes, Ctrl+click *currently* is the same as regular click - both select just the current staff.  I was suggesting we *could* change regular click to select across all staves similarly to key and time signatures, *if* we wanted to fix the consistency problem by having *all* properties affect all staves.

Reply:
Yes, I'm only testing the "BarlineType" property (which is shown as "Style" in the inspector/properties panel).  How do I change it for a single staff (other than dragging from the palette)?

Reply:
As I have been saying, that's the one property that behaves inconsistently - it's the only property that currently applies to all staves.  The only current way to apply the "Style" property to only selected staves is via the palette using Ctrl.

This is why I am pointing out that this setting serves no uniquely useful purpose currently, and that since it's *also* inconsistent with the other properties, I'm totally fine with seeing it removed.  At least right now I don't see a downside.